lib/backend/etcdbk/etcd.go

```diff
@@ -491,7 +512,7 @@ func (b *EtcdBackend) watchEvents(ctx context.Context) error {
 	emitDone := make(chan struct{})
 
 	// watcher must be registered before we initialize the buffer
-	eventsC := b.client.Watch(ctx, b.cfg.Key, clientv3.WithPrefix())
+	eventsC := b.clients.Next().Watch(ctx, b.cfg.Key, clientv3.WithPrefix())
```

Comment:
idea:

If you wanted, to, you could make these calls look like `b.client().Watch` by adding a roundRobinFunc type. But `b.clients.Next()` might be better for its explicitness and ease of access to internals (for closing clients).

```
type roundRobinFunc[T any] func() T

func newRoundRobinFunc[T any](items []T) roundRobinFunc[T] {
	rb := roundRobin[T]{
		ct:    new(atomic.Uint64),
		items: items,
	}
	return rb.Next
}
```

Reply:
I'm sure there's more I could do to polish the `roundRobin` concept, but yeah, given the need for closers I think its always gonna be cleaner as a custom type with fields/methods rather than a closure.